packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -47,39 +51,55 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
       stars.push(this._renderStar(i));
     }
 
+    let ratingContainer: JSX.Element;
+
+    if (this.props.aggregate) {
+      ratingContainer = <div className={ css('ms-Rating-container', styles.container) } role='img' alt={ this.props.rating.toString() } aria-labelledby={ `${this._labelId}-${this.props.rating}` }>
+        { this._getLabel(this.props.rating) }
+        { stars }
+      </div>;
+    } else {
+      ratingContainer = <div className={ css('ms-Rating-container', styles.container) } role='radiogroup' aria-labelledby={ this.props.ariaLabelId }>
+        { stars }
+      </div>;
+    }
+
     return <div className={ css('ms-Rating', this.props.className, {
       ['ms-Rating--large ' + styles.rootIsLarge]: this.props.size === RatingSize.Large
     }) } role='application'>
-      <div className={ css('ms-Rating-container', styles.container) } role='radiogroup' aria-labelledby={ this.props.ariaLabelId }>
-        { stars }
-      </div>
+      { ratingContainer }
     </div>;
   }
 
   private _renderStar(rating: number): JSX.Element {
     const inputId = `${this._id}-${rating}`;
+    const flooredRating = Math.floor(this.state.rating);
+    const starSize = this.props.aggregate && rating > this.state.rating && rating - 1 < this.state.rating ? this.state.rating - flooredRating : 1;
```

Comment:
If we pass rating to getIcon and do the calculation there, it will make the code confusing. Instead, we directly telling the getIcon to render specific size of icon. This makes code easy to understand